### PR TITLE
Add gct

### PR DESCRIPTION
**Pull request by chrisburr**
Checklist:

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by chrisburr** [comment]
@conda-forge/staged-recipes This is ready for review

File: recipes/gct/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    - {{ compiler('cxx') }}
+    - automake
+    - make
+    - pkg-config
+    - libtool
+  host:
+    - openssl =1.0
+    - zlib
+  run:
+    - openssl
```
Comment (ocefpaf) [inline recipes/gct/meta.yaml]: Is the openssl pin needed only on host or do you need it here too?

File: recipes/gct/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    - {{ compiler('cxx') }}
+    - automake
+    - make
+    - pkg-config
+    - libtool
+  host:
+    - openssl =1.0
+    - zlib
+  run:
+    - openssl =1.0
```
Comment (scopatz) [inline recipes/gct/meta.yaml]: Is this the right way to pin openssl?

**Comment by chrisburr** [comment]
Thanks for taking a look 👍

For OpenSSL this can rely on `run_exports` (and I'll rebuild for OpenSSL 1.1 in the feedstock).

**Comment by chrisburr** [comment]
Do you have any objections to me merging this?

File: recipes/gct/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - automake
+    - make
+    - pkg-config
+    - libtool
+  host:
+    - openssl =1.0
```
Comment (ocefpaf) [inline recipes/gct/meta.yaml]: I'm not 100% sure we had `run_exports` back in version `1.0.*`. If only 1.0 works I would try:

```
  host:
    - openssl 1.0.2r
  run:
    - openssl 1.0.2r
```

Note that the pin must be `x.x.x`, so I would no do `=1.0` b/c things may be broken.

File: recipes/gct/meta.yaml [hunk 4]
```diff
@@ -24,9 +24,10 @@ requirements:
     - pkg-config
     - libtool
   host:
-    - openssl =1.0
+    - openssl 1.0.2r
     - zlib
   run:
+    - openssl 1.0.2r
```
Comment (ocefpaf) [inline recipes/gct/meta.yaml]: BTW, `1.0.2r` was just an example. I'm not sure that version is OK for `gct`. Also, do you know it does not work with the latest `1.1.1b`?

Comment (chrisburr) [reply recipes/gct/meta.yaml]: IIRC `gct` already works with any version of OpenSSL.

The issue is I'm currently build a relatively large chain of dependencies and one of them doesn't (`pygsi`). It's going to take a while until that is patched (or maybe dropped) so I was going to add everything with `OpenSSL 1.0.2` then update in the feedstocks.

**Comment by chrisburr** [comment]
Thanks! 🍰

**Pull request merged**